src/components/merge_results/fields/CompareFunding.tsx

```diff
@@ -1,3 +1,71 @@
-export const CompareFunding = () => {
-  return;
+import ArrowForwardIcon from '@mui/icons-material/ArrowForwardIos';
+import RestoreIcon from '@mui/icons-material/Restore';
+import { Divider, Typography } from '@mui/material';
+import { useContext } from 'react';
+import { useFieldArray, useFormContext } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { Funding, Registration } from '../../../types/registration.types';
+import { dataTestId } from '../../../utils/dataTestIds';
+import { MergeResultsWizardContext } from '../MergeResultsWizardContext';
+import { StyledCompareButton } from './CompareFiles';
+import { FundingBox } from './FundingBox';
+
+interface CompareFundingProps {
+  sourceFunding?: Funding;
+  targetFunding?: Funding;
+  matchingTargetFundingIndex?: number;
+}
+export const CompareFunding = ({
+  sourceFunding,
+  targetFunding,
+  matchingTargetFundingIndex = -1,
+}: CompareFundingProps) => {
+  const { t } = useTranslation();
+  const { sourceHeading } = useContext(MergeResultsWizardContext);
+
+  const { control } = useFormContext<Registration>();
+  const { append, remove } = useFieldArray({ name: 'fundings', control });
+
+  const canCopyFunding = !!sourceFunding && !targetFunding;
+  const fundingIsCopied = matchingTargetFundingIndex > -1;
+
+  return (
+    <>
+      <Typography variant="h3" sx={{ display: { xs: 'block', md: 'none' } }}>
+        {sourceHeading}
+      </Typography>
+      <FundingBox funding={sourceFunding} />
+
+      {canCopyFunding && (
+        <StyledCompareButton
+          data-testid={dataTestId.basicData.centralImport.copyValueButton}
+          variant="contained"
+          color="secondary"
+          size="small"
+          endIcon={<ArrowForwardIcon />}
+          onClick={() => append(sourceFunding)}>
+          {t('project.form.add_financing')}
+        </StyledCompareButton>
```

Comment:
Hvis StyledCompareButton er en komponent istedet for en styled component, kan variant, color, size etc være en del av komponenten så det slipper å bli deklarert hver gang Buttonen skal brukes?